core/src/main/scalajs/sttp/client3/AbstractFetchBackend.scala

```diff
@@ -240,63 +245,107 @@ abstract class AbstractFetchBackend[F[_], S <: Streams[S], P](
       .order(original.order)
   }
 
-  protected def handleStreamBody(s: streams.BinaryStream): F[js.UndefOr[BodyInit]]
+  private def sendWebSocket[T, R >: PE](request: Request[T, R]): F[Response[T]] = {
+    val queue = new JSSimpleQueue[F, WebSocketEvent]
+    val ws = new JSWebSocket(request.uri.toString)
+    ws.binaryType = BinaryType
 
-  private lazy val bodyFromResponseAs =
-    new BodyFromResponseAs[F, FetchResponse, Nothing, streams.BinaryStream] {
-      override protected def withReplayableBody(
-          response: FetchResponse,
-          replayableBody: Either[Array[Byte], SttpFile]
-      ): F[FetchResponse] = {
-        val bytes = replayableBody match {
-          case Left(byteArray) => byteArray
-          case Right(file)     => throw new IllegalArgumentException("Replayable file bodies are not supported")
-        }
-        new FetchResponse(bytes.toTypedArray.asInstanceOf[BodyInit], response.asInstanceOf[ResponseInit]).unit
-      }
+    val isOpen = Promise[Unit]()
 
-      override protected def regularIgnore(response: FetchResponse): F[Unit] =
-        transformPromise(response.arrayBuffer()).map(_ => ())
+    ws.onopen = (_: Event) => {
+      isOpen.success(())
+      queue.offer(WebSocketEvent.Open())
+    }
+    ws.onmessage = (event: MessageEvent) => queue.offer(toWebSocketEvent(event))
+    ws.onerror = (_: Event) => {
+      if (!isOpen.isCompleted) isOpen.failure(new ReadException(request, new WebSocketTimeoutException))
```

Comment:
We don't really know if it's a timeout. Could be, could be not. I guess the best we can tell is that the WS was not opened for some reason.

Btw. there's really no way of finding out the error message? :)